### PR TITLE
One click setup

### PR DESCRIPTION
**Pull request by jusasiiv**
One click setup changes for review

### CODE REVIEW

File: blockonomics-woocommerce.php [hunk 1]
```diff
@@ -115,93 +141,116 @@ function add_page()
...
+        {
+            $callback_secret = get_option('blockonomics_callback_secret');
+            $callback_url = WC()->api_request_url('WC_Gateway_Blockonomics');
+            $callback_url = add_query_arg('secret', $callback_secret, $callback_url);
+            return $callback_url;
+        }
+
+        function get_total_received()
+        {
+            $blockonomics_orders = get_option('blockonomics_orders');
```
Comment (jusasiiv) [inline blockonomics-woocommerce.php]: What if there are under or over payments? Using this we only see the requested amounts

File: blockonomics-woocommerce.php [hunk 2]
```diff
@@ -98,9 +98,35 @@ function woocommerce_handle_blockonomics_return()
...
+            // temp wallet will be used with temp api key
+            if ($api_key == null)
+            {
+                generate_secret();
+                $callback_url = get_callback_url();
+                $response = $blockonomics->get_temp_api_key($callback_url);
+
+                if ($response->response_code != 200)
+                {
+                    $message = __('Error while generating temporary APIKey: '. $response->message, 'blockonomics-bitcoin-payments');
```
Comment (jusasiiv) [inline blockonomics-woocommerce.php]: What if error does not have message? For example 401 or other

**Comment by blockonomics** [comment]
1. Temp API key creation should be in activation function

File: blockonomics-woocommerce.php [hunk 3]
```diff
@@ -98,9 +98,35 @@ function woocommerce_handle_blockonomics_return()
...
+            }
+
+            $api_key = $blockonomics->get_api_key();
             
-            generate_secret();
-            register_setting('blockonomics_g', 'blockonomics_gen_callback', 'gen_callback');
+            // get_api_key() will return api key or temp api key
+            // if both are null, generate new blockonomics guest account with temporary wallet
+            // temp wallet will be used with temp api key
+            if ($api_key == null)
```
Comment (blockonomics) [inline blockonomics-woocommerce.php]: I don't like comparison with `null` .. can we use if (!$api_key).  It should cover empty API key case also

File: blockonomics-woocommerce.php [hunk 4]
```diff
@@ -109,99 +135,133 @@ function add_page()
...
                 }
                 else
                 {
                     $message = __('Congrats ! Setup is all done', 'blockonomics-bitcoin-payments');
-                    $type = 'updated';
-                    add_settings_error('option_notice', 'option_notice', $message, $type);
+                    display_admin_message($message, 'updated');
+
+                    // Only make withdraw if temp api key is not null
+                    if(get_option('blockonomics_temp_api_key') != null)
```
Comment (blockonomics) [inline blockonomics-woocommerce.php]: Lets avoid null comparisons

File: php/Blockonomics.php [hunk 5]
```diff
@@ -42,36 +54,50 @@ public function new_address($api_key, $secret, $reset=false)
...
+        $body = json_encode(array('callback' => $callback_url));
+        $response = $this->post($url, '', $body);
+        $responseObj = json_decode(wp_remote_retrieve_body($response));
+        $responseObj->{'response_code'} = wp_remote_retrieve_response_code($response);
+        return $responseObj;
+    }
+
+    public function make_withdraw()
+    {
+        $api_key = get_option("blockonomics_api_key");
```
Comment (blockonomics) [inline php/Blockonomics.php]: use api_key from object instance

File: php/Blockonomics.php [hunk 6]
```diff
@@ -42,36 +54,50 @@ public function new_address($api_key, $secret, $reset=false)
...
+        $temp_api_key = get_option('blockonomics_temp_api_key');
```
Comment (blockonomics) [inline php/Blockonomics.php]: Do some sanity checks.. if temp_api_key is empty or api_key doesn't exist ... then do nothing

File: php/WC_Gateway_Blockonomics.php [hunk 7]
```diff
@@ -208,6 +206,12 @@ public function check_blockonomics_callback()
                         $wc_order->add_order_note(__('Payment completed', 'blockonomics-bitcoin-payments'));
                         $wc_order->payment_complete($order['txid']);
                     }
+                    // Keep track of funds in temp wallet
+                    if(get_option('blockonomics_temp_api_key')) {
+                        $current_temp_amount = get_option('temp_withdraw_amount');
```
[the remark this answers is not in the document]
Comment (jusasiiv) [reply php/WC_Gateway_Blockonomics.php]: `get_option()` returns False if there is no value. PHP handles False as 0 if making calculations so here we can assume no option = 0

File: blockonomics-woocommerce.php (hunk 2 above)
Comment (blockonomics) [reply blockonomics-woocommerce.php]: if get_temp_api_key doesn't work. Its probably a critical error. Very less chance of happening

File: php/Blockonomics.php [hunk 8]
```diff
@@ -42,36 +54,66 @@ public function new_address($api_key, $secret, $reset=false)
...
+        $responseObj = json_decode(wp_remote_retrieve_body($response));
+        $responseObj->{'response_code'} = wp_remote_retrieve_response_code($response);
+        return $responseObj;
+    }
+
+    public function make_withdraw()
+    {
+        $api_key = $this->api_key;
+        $temp_api_key = get_option('blockonomics_temp_api_key');
+        if (!$api_key || !$temp_api_key) {
```
Comment (blockonomics) [inline php/Blockonomics.php]: Also check if `$temp_api_key != $api_key`

File: php/Blockonomics.php [hunk 9]
```diff
@@ -8,16 +8,28 @@ class Blockonomics
...
 
     public function __construct()
     {
+        $this->api_key = $this->get_api_key();
+    }
+
+    public function get_api_key()
+    {
+        $api_key = get_option("blockonomics_api_key");
+        if ($api_key == null)
```
Comment (blockonomics) [inline php/Blockonomics.php]: Lets avoid null comparison

**Pull request merged**